app/shop.py

```diff
@@ -0,0 +1,47 @@
+from math import sqrt
+from datetime import datetime
+
+
+class Shop:
+    def __init__(self, name: str, location: tuple, products: dict) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def calculate_trip_cost(
+            self,
+            customer_location: tuple,
+            fuel_price: float,
+            fuel_consumption: float
+    ) -> tuple:
+        distance = sqrt((self.location[0] - customer_location[0]) ** 2
+                        + (self.location[1] - customer_location[1]) ** 2)
+        return round(
+            distance * 2 * fuel_consumption / 100 * fuel_price, 2
+        ), distance
+
+    def purchase(self, customer_name: str, product_cart: dict) -> tuple:
+        total_cost = 0
+        specific_time = datetime(2021, 4, 1, 12, 33, 41)
+        receipt = [
+            f"Date: {specific_time.strftime("%m/%d/%Y %H:%M:%S")}",

```

Comment:
The date in the receipt is hardcoded to a specific time. Consider using the current date and time to make the receipt more dynamic and realistic.